src/models/event-timeline-set.js

```diff
@@ -180,13 +191,37 @@ EventTimelineSet.prototype.resetLiveTimeline = function(backPaginationToken, flu
             events.push(evMap[evtype][stateKey]);
         }
     }
+
+    // Use those events to initialise the state of the new live timeline
     newTimeline.initialiseState(events);
 
+    const freshEndState = newTimeline._endState;
+    // Now clobber the end state of the new live timeline with that from the
+    // previous live timeline. It will be identical except that we'll keep
+    // using the same RoomMember objects for the 'live' set of members with any
+    // listeners still attached
+    newTimeline._endState = this._liveTimeline._endState;
+
+    // If we're not resetting all timelines, we need to fix up the old live timeline
+    if (!resetAllTimelines) {
+        // Firstly, we just stole the old timeline's end state, so it needs a new one.
+        // Just swap them around and give it the one we just generated for the
+        // new live timeline.
+        this._liveTimeline._endState = freshEndState;
+
+        // Now set the forward pagination token on the old live timeline
+        // so it can be forward-paginated.
+        this._liveTimeline.setPaginationToken(
```

Comment:
Nit: A helpfully named `oldTimeline` variable to complement `newTimeline` would help readability on this.